snowtrick/assets/app.js

```diff
@@ -6,7 +6,116 @@
  */
 
 // any CSS you import will output into a single css file (app.css in this case)
-import './styles/app.css';
+import './styles/app.scss';
 
 // start the Stimulus application
-import './bootstrap';
+
+document.addEventListener('DOMContentLoaded', () => {
+// Functions to open and close a modal
+    function openModal($el) {
+        $el.classList.add('is-active');
+    }
+
+    function closeModal($el) {
+        $el.classList.remove('is-active');
+    }
+
+    function closeAllModals() {
+        (document.querySelectorAll('.modal') || []).forEach(($modal) => {
+            closeModal($modal);
+        });
+    }
+
+// Add a click event on buttons to open a specific modal
+    (document.querySelectorAll('.js-modal-trigger') || []).forEach(($trigger) => {
+        const modal = $trigger.dataset.target;
+        const $target = document.getElementById(modal);
+
+        $trigger.addEventListener('click', () => {
+            openModal($target);
+        });
+    });
+
+// Add a click event on various child elements to close the parent modal
+    (document.querySelectorAll('.modal-background, .modal-close, .modal-card-head .delete, .modal-card-foot .button') || []).forEach(($close) => {
+        const $target = $close.closest('.modal');
+
+        $close.addEventListener('click', () => {
+            closeModal($target);
+        });
+    });
+});
+
+// add event on trigger button
+function activeTrashButton() {
+    const trash_btn = document.querySelectorAll(".trash_btn");
+    trash_btn.forEach((trash) => {
+        trash.addEventListener('click', (event) => {
+            let myInit = {
+                method: 'POST',
+                headers: {'content-type': 'application/json'}
+            };
```

Comment:
Codacy has a fix for the issue: [Replace `············` with `······`](https://app.codacy.com/gh/trousse/OC_Projet6/pullRequest?prid=10057695)
```suggestion
      };
```